src/teehr/classes/duckdb_base.py

```diff
@@ -53,20 +54,16 @@ def query(
         """Run an SQL query against the class's database.
 
         Return formats include:
-
         * A pandas dataframe (format='df')
         * Results printed to the screen (format='raw')
-        * A DuckDBPyRelation, a symbolic representation of the SQL query
-          (format='relation').
         """
-        if format == "df":
-            return self.con.sql(query).df()
-        elif format == "raw":
-            return self.con.sql(query).show()
-        # elif format == "relation":
-        #     return self.con.sql(query)
-        self.con.sql(query)
-        return None
+        with duckdb.connect() as con:
+            resp = con.cursor().sql(query)
```

Comment:
Just curious about the use of the cursor here, did you find it was necessary to use to solve a specific problem?

Reply:
ah, no good catch, tried it when i was fixing the failing tests and is slipped through when i was cleaning up.